controllers/storagecluster/uninstall_reconciler.go

```diff
@@ -326,8 +341,9 @@ func (r *StorageClusterReconciler) deleteResources(sc *ocsv1.StorageCluster) (re
 
 	objs := []resourceManager{
 		&ocsExternalResources{},
-		&ocsProviderServer{},
 		&ocsNoobaaSystem{},
+		&storageClient{},
```

Comment:
It should be prioritized at the top, being the first to be deleted before anything else. Otherwise, we might end up deleting other items first and then requesting its deletion, which is not the desired sequence. This particular item should be removed first, and only after that, we should proceed to delete storage cluster resources.

Reply:
Agreed with Nitin. Also, why ar eyou moving the provider server uninstall down? I would think it should stay up where it was.

Reply:
I moved it down as to ensure we delete storageclient first which eventually removes storageconsumer and provider server fails to delete if storageconsumer exists.

Reply:
> It should be prioritized at the top
- done.